run/test/setup.ts

```diff
@@ -47,52 +48,44 @@ describe('setup', function() {
   });
 
   it('should type-check keyof sources and sinks in main and drivers', function() {
-    type Sources = {
-      str: Stream<string>;
-      obj: Stream<object>;
-    };
-
-    function app(sources: Sources) {
+    function app(sources: {str: Stream<string>; obj: Stream<object>}) {
       return {
         str: sources.str.take(1).startWith('a'), // good
         // str: sources.obj.mapTo('good'), // good
-        // strTYPO: sources.str.take(1).startWith('a'), // bad
+        // strTYPO: xs.of('a'), // good (but in older versions was bad)
```

Comment:
Shouldnt this be bad if the key is not in the drivers object?

Reply:
So, there are a couple of real use cases where I take a ready `main` function and plug it with my drivers, and I choose to ignore a certain sink because I'm not going to use it.

That said, I think you have a point, maybe we could support that case by forcing the user to give a noop driver for that channel. But the other way around (drivers provide some sources but `main` never use them) doesn't need this kind of check.

Reply:
Yeah, i think everything returned by main should be in the drivers object, but not the other way round